.circleci/config.yml

```diff
@@ -0,0 +1,124 @@
+#   * TERMINUS_SOURCE_ENVIRONMENT
+#
+version: 2
+jobs:
+  build:
+    working_directory: /var/www/code
+    docker:
+      - image: lastcallmedia/php:7.0-dev
+    steps:
+      - checkout
+      - restore_cache:
+          name: Restore Yarn cache
+          keys:
+            - site-yarn-v1-{{ checksum "yarn.lock" }}
+            - site-yarn-v1-
+      - run: {name: "Yarn install", command: "yarn install --pure-lockfile" }
+      - save_cache:
+          name: Save Yarn cache
+          key: site-yarn-v1-{{ checksum "yarn.lock" }}
+          paths: [ node_modules ]
+      - restore_cache:
+          name: Restore Composer cache
+          keys:
+            - site-composer-v1-{{ checksum "composer.lock" }}
+            - site-composer-v1-
+      - run: {name: 'Composer install', command: 'composer install'}
+      - save_cache:
+          name: Save Composer cache
+          key: site-composer-v1-{{ checksum "composer.lock" }}
+          paths: [ vendor, web/core, web/modules/contrib, web/themes/contrib ]
+      - run: {name: "Execute build", command: "gulp build" }
+      - persist_to_workspace:
+          root: /var/www
+          paths: [code]
+
+  test:
+    working_directory: /var/www/code
+    docker:
+      - image: lastcallmedia/php:7.0-dev # Primary image
+        environment:
+          DOCKER_ENV: ci
+          APACHE_DOCROOT: /var/www/code/web
+          MYSQL_USER: circle
+          MYSQL_PASSWORD: circle
+          MYSQL_DATABASE: circle
+          MYSQL_HOST: 127.0.0.1
+      - image: mysql:5.6
+        environment:
+          MYSQL_USER: circle
+          MYSQL_PASSWORD: circle
+          MYSQL_DATABASE: circle
+          MYSQL_RANDOM_ROOT_PASSWORD: 1
+      - image: selenium/standalone-chrome
+        name: selenium
+    steps:
+      - attach_workspace: {at: /var/www}
+      - run: {name: 'Source environment variables', command: 'echo "source /var/www/code/docker/drupal.env" >> $BASH_ENV'}
+      - run: {name: 'Setup artifact directories', command: 'mkdir /tmp/junit /tmp/artifacts && chown $APACHE_RUN_USER:$APACHE_RUN_GROUP /tmp/junit'}
+      # Run linting steps up front.  If these fail, they should provide fast feedback.
+      - run: {name: 'ESLint', command: 'node_modules/.bin/eslint -f junit . > /tmp/junit/eslint.xml'}
+      - run: {name: 'PHPCS', command: 'vendor/bin/phpcs --report-junit=/tmp/junit/phpcs.xml --report-summary'}
+      - run: {name: 'Mannequin Snapshot', command: 'vendor/bin/mannequin snapshot -o /tmp/artifacts/mannequin'}
+      - run: {name: 'Start Apache', command: '/usr/local/bin/apache2-foreground-enhanced', background: true}
+      - run: {name: 'Set up directories', command: 'mkdir web/sites/simpletest web/sites/default/files && chown -R $APACHE_RUN_USER:$APACHE_RUN_GROUP web/sites/default/files web/sites/simpletest'}
+      - run: {name: 'Set DB cache invalidator', command: 'echo "$(date +%Y-%m)-$TERMINUS_MACHINE_TOKEN-$TERMINUS_SITE-$TERMINUS_SOURCE_ENV" > /tmp/cache-indicator' }
```

Comment:
Yeah, interesting thing here.  `bin/refresh-local-pantheon` has some logic in it to fetch the filename of the most recent backup using `terminus backup: info` every time it's run, then only download that file if it doesn't exist in the cache directory.  So even though we cache for a month (just the easiest denominator because I didn't want to run into issues with the time Pantheon takes the backup), we're always getting the most recent backup. If you can think of ways to clarify that, I'm all for it.  I'd rather keep our `config.yml` as self-documenting and comment free as we can, so is there a way we can use the step name to make it more clear?

Reply:
Actually, I thought about this, and I think you're right. It's not great to hang onto a month of old backups when we don't need to, and the cache indicator calculation isn't immediately obvious.  I'll switch it.  Thanks!